poseidon/poseidon.go

```diff
@@ -157,16 +140,16 @@ func PoseidonHash(inp [T]*big.Int) (*big.Int, error) {
 	return state[0], nil
 }
 
-// Hash performs the Poseidon hash over a *big.Int array
+// Hash performs the Poseidon hash over a ff.Element array
 // in chunks of 5 elements
-func Hash(arr []*big.Int) (*big.Int, error) {
-	if !utils.CheckBigIntArrayInField(arr, constants.Q) {
+func Hash(arr []*ff.Element) (*ff.Element, error) {
```

Comment:
Have you considered keeping `*big.Int` in the public API (since they are part of go's standard library) and only using `*ff.Element` internally?

Reply:
Cool, yes! Done changes, now everything is exposed with `*big.Int`, and inside `PoseidonHash` function it converts to `*ff.Element`